File: leetCode/two-pointers/language/python/392.is-subsequence.py
def isSubsequence(s: str, t: str) -> bool:
    if not s:
        return True
    if not t:
        return False
    i = 0
    for c in t:
        if c == s[i]:
            i += 1
            if i == len(s):
                return True
    return False


if __name__ == "__main__":
    print(isSubsequence("abc", "ahbgdc"))

File: leetCode/two-pointers/language/rust/392.is-subsequence.rs
fn main() {
    println!(
        "{}",
        is_subsequence("abc".to_string(), "ahbgdc".to_string())
    );
}

fn is_subsequence(s: String, t: String) -> bool {
    let mut s = s.chars().collect::<Vec<_>>();
    let mut t = t.chars().collect::<Vec<_>>();
    let mut i = 0;
    for j in 0..t.len() {
        if i == s.len() {
            return true;
        }
        if s[i] == t[j] {
            i += 1;
        }
    }
    i == s.len()
}
